ray-operator/controllers/ray/rayjob_controller.go

```diff
@@ -910,6 +936,43 @@ func updateStatusToSuspendingIfNeeded(ctx context.Context, rayJob *rayv1.RayJob)
 	return true
 }
 
+func (r *RayJobReconciler) checkSubmitterAndUpdateStatusIfNeeded(ctx context.Context, rayJob *rayv1.RayJob, submissionMode rayv1.JobSubmissionMode) (shouldUpdate, isK8sJobFinished bool, err error) {
```

Comment:
Initialize the logger at the beginning of the function so that we don't need to initialize it in both `case rayv1.K8sJobMode:` and `default:`.